docs/adr/0004-link.md

```diff
@@ -0,0 +1,74 @@
+# Link
+
+* Status: accepted
+* Deciders: @dunglas, @soyuka, @alanpoulain
+
+Implementation: [#4536][pull/4536]
+
+## Context and Problem Statement
+
+The [URI Variables](0003-uri-variables.md) ADR introduces a new `UriVariable` POPO.
+In GraphQL, having URI variables make no sense: this object needs either an alias or needs to be named differently.
+
+## Considered Options
+
+* Create a `Traverser` alias for GraphQL.
+* Rename `UriVariable` to `Link`.
+
+## Decision Outcome
+
+We chose to rename `UriVariable` to `Link` in order to simplify the codebase.
+However the `uriVariables` parameter in the REST operations will not be renamed since it makes sense to have this name.
+GraphQL operations don't need to have links at the operation level, a `Link` attribute on the property will be used instead if necessary (the main use case is when an `inverseProperty` is necessary).
+
+### Example
+
+```php
+<?php
+
+#[Query]
+#[Get('/companies/{companyId}/employees', uriVariables: [
+    'companyId' => new Link(
+        targetClass: Company::class,
+        identifiers: ['id'],
+        property: 'company'
```

Comment:
inverseProperty: 'employees' ?

Reply:
The `company` property is in the `Employee` class, it seems alright to me, isn't it?